src/sciline/task_graph.py

```diff
@@ -49,11 +49,19 @@ def compute(
             Optional list of keys to compute. This can be used to override the keys
             stored in the graph instance. Note that the keys must be present in the
             graph as intermediate results, otherwise KeyError is raised.
+
+        Returns
+        -------
+        If ``keys`` is a single type, returns the single result that was computed.
+        If ``keys`` is a tuple of types, returns a dictionary with type as keys
+        and the corresponding results as values.
```

Comment:
As @jl-wynen pointed out: mypy will not be happy? Does adding an `@overload` help?

Reply:
It is `Any` so it shouldn't matter, no...?

Reply:
True. Can we split it into `Any` and `Dict[type, Any]` now?

Reply:
You mean like this...?
```python
def compute(self, tp: type | Tuple[type, ...] | Item[T]) -> Any | Dict[Type[T], T]: ...
```

Reply:
Isn't it already like this in `pipeline.Pipeline`? and `task_graph.TaskGraph` can also have them but it doesn't really work with `None` since the returned type depends on the `self._keys` attribute type, not the argument type.

Reply:
Right.